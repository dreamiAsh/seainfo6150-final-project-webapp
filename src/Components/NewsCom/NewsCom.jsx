import React, {useEffect, useState} from "react";
import styles from "./NewsCom.module.css";
import { Link } from "react-router-dom";
import NewsItem from "../NewsItem/NewsItem";

const NewsCom = (props) => {

    const [fetchedData, setFetchedData] = useState({});
    useEffect(() => {
      const fetchData = async () => {
        // put data fetching code here!
        // performs a GET request
        const response = await fetch(
          "https://demo9533681.mockable.io/popularList"
        );
        const responseJson = await response.json();
        setFetchedData(responseJson);
      };
      fetchData();
    }, [fetchedData]);

    const myList = Object.values(fetchedData)
    const mappedList = myList.map((listItem) => (
      <NewsItem listItem={listItem}/>
    ))

    return (
        <div className={styles.box}>
            <ul>
                {mappedList}
            </ul>
        </div>
    );
};

export default NewsCom;
