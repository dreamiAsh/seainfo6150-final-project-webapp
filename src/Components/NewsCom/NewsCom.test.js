import React from "react";
import { render, fireEvent } from "@testing-library/react";
import NewsCom from "./NewsCom.jsx";

it("renders the NewsCom component correctly", () => {
  const { container } = render(<NewsCom />);
  expect(container).toMatchSnapshot();
});