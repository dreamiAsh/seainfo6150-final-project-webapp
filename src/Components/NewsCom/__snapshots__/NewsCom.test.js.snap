// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the NewsCom component correctly 1`] = `
<div>
  <div
    class="box"
  >
    <ul />
  </div>
</div>
`;
